core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineAnalyzer.java

```diff
@@ -223,40 +223,40 @@ private static void validateArguments(FunctionSpecification function)
         }
     }
 
-    private static Optional<String> getLanguage(FunctionSpecification function)
+    private static Optional<Identifier> getLanguage(FunctionSpecification function)
     {
         List<LanguageCharacteristic> language = function.getRoutineCharacteristics().stream()
                 .filter(LanguageCharacteristic.class::isInstance)
                 .map(LanguageCharacteristic.class::cast)
-                .collect(toImmutableList());
+                .toList();
```

Comment:
why not `.collect(toImmutableList());` (as it used to be)?

(same below)

if this change is desired, but is a refactor, let's put it in a refactor commit, not in a "Fix ..." commit